MIP/move-bridge-architecture/README.md

```diff
@@ -0,0 +1,165 @@
+What we want is some _atomicity_ between the steps: if the user locks their funds, then either the corresponding minting transaction occurs, or if it does not (and we may set a time bound), the funds are returned to the user on L1.
+Another source of difficulty is to make sure that only the user `l2acc` can redeem the wrapped L2\$MOVE tokens. i.e. they are not credited to another user.
+
+All these problems have been thoroughly studied and bridges have been in operation for several years. However hacks related to bridges account for more than 1/3 of the total hacks values which tends to indicate that bridges are vulnerable, frequently attacked, and should be designed carefully. Infamous attacks are the two Ronin bridge attacks [2022: Crypto Hackers Exploit Ronin Network for $615 Million](https://www.bankinfosecurity.com/crypto-hackers-exploit-ronin-network-for-615-million-a-18810), and [2024: Ronin Bridge Paused, Restarted After $12M Drained in Whitehat Hack](https://www.coindesk.com/tech/2024/08/06/ronin-bridge-paused-after-9m-drained-in-apparent-whitehat-hack/), and the Nomad bridge attack [August 2022: Hack Analysis: Nomad Bridge ($192M)](https://medium.com/immunefi/hack-analysis-nomad-bridge-august-2022-5aa63d53814a).
+Some solutions [XChainWatcher](https://arxiv.org/abs/2410.02029) rely on monitoring bridges and detect attacks.
+
+Designing a safe bridge is a hard problem.
+
+## Specification
+
+<!--
+
+  The Specification section should describe the syntax and semantics of any new feature. The specification should be detailed enough to allow competing, interoperable implementations.
+
+  It is recommended to follow RFC 2119 and RFC 8170. Do not remove the key word definitions if RFC 2119 and RFC 8170 are followed.
+
+  TODO: Remove this comment before finalizing
+-->
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+## Reference Implementation
+
+### Bridging from L1 to L2
+
+The contracts involved are:
+
+- Solidity contract [AtomicBridgeInitiatorMOVE.sol](https://github.com/movementlabsxyz/movement/blob/main/protocol-units/bridge/contracts/src/AtomicBridgeInitiatorMOVE.sol) on L1,
+- module [atomic_bridge_counterparty.move](https://github.com/movementlabsxyz/aptos-core/blob/061155119258caab512aec6aa860b086e5f312e0/aptos-move/framework/aptos-framework/sources/atomic_bridge.move#L1163) on L2.
+
+The current implementation is a _lock-mint_ bridge. The user locks their L1\$MOVE tokens in the `AtomicBridgeInitiatorMOVE` contract, and the `atomic_bridge_counterparty.move` module mints the corresponding L2\$MOVE tokens.
+An overview of the _happy_ path for normal operation is as follows:
+
+![alt text](L1ToL2.png)
+
+> [!WARNING]
+If `timeLockL2` is larger than `timeLockL1`, the following scenario can happen:
```

Comment:
i am not sure i understand the picture but it seems the problem occurs already if lockBridgeTransferAssets happens late enough. No need to have timeLockL2 larger than timeLockL1